packages/appkit/src/client.ts

```diff
@@ -1190,9 +1190,12 @@ export class AppKit {
     })
     provider.onConnect(async user => {
       const namespace = ChainController.state.activeChain as ChainNamespace
+
+      const connector = this.getConnectors().find(c => c.id === ConstantsUtil.CONNECTOR_ID.AUTH)
+
       this.syncProvider({
         type: UtilConstantsUtil.CONNECTOR_TYPE_AUTH as ConnectorType,
-        provider,
+        provider: namespace === ConstantsUtil.CHAIN.SOLANA ? connector?.provider : provider,
```

Comment:
With solana it's a bit tricky since we already add auth connector on the adapter level so technically we need to get it from there as well. With other adapters it's fine since we use the same auth provider that's coming from the appkit client.

Reply:
but the provider should be the same provider 🤔 
they are pointing to the same instance

Reply:
For some reason it doesn't get the auth provider from solana adapter [here](https://github.com/reown-com/appkit/blob/main/packages/appkit/src/client.ts#L1195) because we create two auth providers one in appkit client and other in solana adapter. 

So what would happen is when user refreshes the page and was connected we'd not get the solana auth provider, but rather appkit client auth provider. This shouldn't happen because auth provider from solana adapter needs to be set which comes from [here](https://github.com/reown-com/appkit/blob/main/packages/adapters/solana/src/client.ts#L98-L107).

Reply:
They should both be pointing to the same provider, that's very weird. Have you tried getting the singleton instance directly instead?